rust/tool/src/install.rs

```diff
@@ -281,3 +316,56 @@ fn ensure_parent_dir(path: &Path) {
         fs::create_dir_all(parent).ok();
     }
 }
+
+/// Determine if a newer systemd-boot version is available.
+///
+/// "Newer" can mean
+///   (1) no file exists at the destination,
+///   (2) the file at the destination is malformed,
+///   (3) a binary with a higher version is available.
+fn newer_systemd_boot(from: &Path, to: &Path) -> Result<bool> {
+    // If the file doesn't exists at the destination, it should be installed.
+    if !to.exists() {
+        return Ok(true);
+    }
+
+    // If the version from the source binary cannot be read, something is irrecoverably wrong.
+    let from_version = systemd_boot_version(from)
+        .with_context(|| format!("Failed to read systemd-boot version from {from:?}."))?;
+
+    // If the version cannot be read from the destination binary, it is malformed. It should be
+    // forcibly reinstalled.
+    let to_version = match systemd_boot_version(to) {
+        Ok(version) => version,
+        _ => return Ok(true),
+    };
+
+    Ok(from_version > to_version)
+}
+
+/// Read the version of a systemd-boot binary from its `.osrel` section.
+///
+/// The version is parsed into a f32 because systemd does not follow strict semver conventions. A
+/// f32, however, should parse all systemd versions and enables usefully comparing them.
```

Comment:
nit: It is a bit of a hack though. ;)

Reply:
Oh yes big time. Do you have a better solution? Or should I just make this clear in the comment as well?

Reply:
If you make it clear in the comments this is fine for me. It does the job.

If someone wants to they can just add a small parser from `String` to `(u32, u32)`.

Reply:
I'll do this in a separate PR.